sdk/src/main/java/com/mongodb/stitch/android/CustomBsonConverter.java

```diff
@@ -0,0 +1,92 @@
+package com.mongodb.stitch.android;
+
+import org.bson.BSONException;
+import org.bson.BsonArray;
+import org.bson.BsonBoolean;
+import org.bson.BsonDocument;
+import org.bson.BsonDouble;
+import org.bson.BsonElement;
+import org.bson.BsonInt64;
+import org.bson.BsonInvalidOperationException;
+import org.bson.BsonNull;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.Document;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by jasonflax on 11/21/17.
+ */
+
+public class CustomBsonConverter {
+    public static BsonArray fromArray(Object[] array) {
+        BsonArray bsonArray = new BsonArray();
```

Comment:
Idk how big we are on finals, but it seems like everywhere else we do it, so maybe add them? I'm only commenting on it here but could be applied to whole PR. :)